apps/minifront/src/components/header/menu/provider.tsx

```diff
@@ -0,0 +1,79 @@
+import { cn } from '@penumbra-zone/ui/lib/utils';
+import * as NavigationMenu from '@radix-ui/react-navigation-menu';
+import { getChainId } from '../../../fetchers/chain-id';
+import { useCallback, useEffect, useState } from 'react';
+import { getPraxManifest, getPraxOrigin } from '@penumbra-zone/client/prax';
+import { itemStyle, triggerStyle, dropdownStyle, linkStyle } from './nav-style';
+
+export const ProviderMenu = () => {
+  const [chainId, setChainId] = useState<string | undefined>();
+  const [providerManifest, setProviderManifest] = useState<ProviderManifest>();
+  const [providerOrigin] = useState(getPraxOrigin());
```

Comment:
[question](https://conventionalcomments.org/): I see the origin is used only for the icons. Can we move the origin field to the manifest or make the `icons` field in the manifest a full URL to support future wallets?

Reply:
the manifest is a chrome extension manifest, which contains no origin field

in prod the origin is the chrome extension id, plus the chrome-extension protocol segment. this is chosen by the app at wallet connection

currently minifront selects prax only so it is a literal. prax is currently special cased within the client package but this ahould be changed. we should soon implement a wallet selection ui and then origin will come from the global injected object

Reply:
in prod the manifest contains id but not in dev, as it simply uses the key